src/CMakeLists.txt

```diff
@@ -3,27 +3,61 @@
 # Original listfile by Daniel Richard G. <skunk@iSKUNK.ORG>
 # This file is in the public domain.
 #
+# Changes 01/20/15 J Kyle Medley
+#
 
 IF(RAPTOR_PARSER_RDFA)
 	INCLUDE_DIRECTORIES(BEFORE ${CMAKE_SOURCE_DIR}/librdfa)
 ENDIF(RAPTOR_PARSER_RDFA)
 
+# ** Parser sources **
+
+# XML parser enabled
 IF(RAPTOR_PARSER_RDFXML)
 	SET(raptor_parser_rdfxml_sources raptor_rdfxml.c)
 	SET(raptor_libxml_libs ${LIBXML2_LIBRARIES})
 ENDIF(RAPTOR_PARSER_RDFXML)
+
+# N triples parser enabled
 IF(RAPTOR_PARSER_NTRIPLES OR RAPTOR_PARSER_NQUADS)
 	SET(raptor_parser_ntriples_nquads_sources ntriples_parse.c raptor_ntriples.c)
 ENDIF(RAPTOR_PARSER_NTRIPLES OR RAPTOR_PARSER_NQUADS)
+
+# Turtle parser enabled
 IF(RAPTOR_PARSER_TURTLE OR RAPTOR_PARSER_TRIG)
-	SET(raptor_parser_turtle_trig_sources
-		turtle_common.h
-		turtle_lexer.c
-		turtle_lexer.h
-		turtle_parser.c
-		turtle_parser.h
-	)
+  SET(raptor_parser_turtle_trig_sources
+#     turtle_common.h
+    ${CMAKE_CURRENT_BINARY_DIR}/turtle_lexer.c
+#     turtle_lexer.h
+    ${CMAKE_CURRENT_BINARY_DIR}/turtle_parser.c
+#     turtle_parser.h
+  )
+
+  # Generate the turtle parser
+  ADD_CUSTOM_TARGET(turtle_tables_tgt DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/turtle_parser.tab.c)
+  ADD_CUSTOM_COMMAND(OUTPUT turtle_parser.tab.c
+    COMMAND ${BISON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/turtle_parser.y
+    DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/turtle_parser.y)
+
+  ADD_CUSTOM_TARGET(turtle_parser_tgt DEPENDS turtle_tables_tgt ${CMAKE_CURRENT_BINARY_DIR}/turtle_parser.c ${CMAKE_CURRENT_BINARY_DIR}/turtle_parser.h)
+  ADD_CUSTOM_COMMAND(OUTPUT turtle_parser.c
+    COMMAND ${PERL_EXECUTABLE} ${CMAKE_SOURCE_DIR}/scripts/fix-bison.pl turtle_parser.tab.c > turtle_parser.c
```

Comment:
won't this line need `${CMAKE_CURRENT_BINARY_DIR}/turtle_parser.c` like the argument above?

same question for the other file references in the `ADD_CUSTOM_COMMAND()` calls here and below